xchange-stream-gemini/src/main/java/info/bitrich/xchangestream/gemini/dto/GeminiOrderbook.java

```diff
@@ -1,50 +1,50 @@
 package info.bitrich.xchangestream.gemini.dto;
 
-import static org.knowm.xchange.gemini.v1.GeminiAdapters.adaptOrders;
-
 import java.math.BigDecimal;
-import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
 import java.util.List;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
 import org.knowm.xchange.currency.CurrencyPair;
 import org.knowm.xchange.dto.Order;
 import org.knowm.xchange.dto.marketdata.OrderBook;
 import org.knowm.xchange.dto.trade.LimitOrder;
 
 /** Created by Lukas Zaoralek on 15.11.17. */
 public class GeminiOrderbook {
-  private final Map<BigDecimal, GeminiLimitOrder> asks;
-  private final Map<BigDecimal, GeminiLimitOrder> bids;
+  private final Map<Double, GeminiLimitOrder> asks;
+  private final Map<Double, GeminiLimitOrder> bids;
```

Comment:
why are you downgrading this do `Double`?

`level.getPrice().doubleValue()` can lead to lost precision

leave `BigDecimal`

Reply:
yes, I was dealing with the horrible performance and trying different things.  BigDecimal is better